service/src/test/java/bio/terra/workspace/app/configuration/external/controller/ControlledGcpResourceApiControllerConnectedTest.java

```diff
@@ -121,124 +118,92 @@ public void cloneBucket_copyNothing() throws Exception {
-            .destinationWorkspaceId(destWorkspaceId)
-            .cloningInstructions(cloningInstructions)
-            .name(TestUtils.appendRandomNumber("bucket-clone"))
-            .jobControl(new ApiJobControl().id(UUID.randomUUID().toString()));
-    String serializedResponse =
-        mockMvc
-            .perform(
-                addJsonContentType(
-                    addAuth(
-                        post(CLONE_CONTROLLED_GCP_GCS_BUCKET_FORMAT.formatted(
-                                sourceWorkspaceId, sourceResourceId))
-                            .content(objectMapper.writeValueAsString(request)),
-                        userAccessUtils.defaultUserAuthRequest())))
-            .andExpect(status().is(HttpStatus.SC_ACCEPTED))
-            .andReturn()
-            .getResponse()
-            .getContentAsString();
-    return objectMapper.readValue(serializedResponse, ApiCloneControlledGcpGcsBucketResult.class);
-  }
-
-  private ApiCloneControlledGcpGcsBucketResult getCloneControlledGcsBucketResult(
-      UUID workspaceId, UUID jobId) throws Exception {
-    String serializedResponse =
-        mockMvc
-            .perform(
-                addJsonContentType(
-                    addAuth(
-                        get(
-                            CLONE_RESULT_CONTROLLED_GCP_GCS_BUCKET_FORMAT.formatted(
-                                workspaceId.toString(), jobId.toString())),
-                        userAccessUtils.defaultUserAuthRequest())))
-            // Returns 200 if flight is done, 202 if flight is running.
-            .andExpect(status().is2xxSuccessful())
-            .andReturn()
-            .getResponse()
-            .getContentAsString();
-    return objectMapper.readValue(serializedResponse, ApiCloneControlledGcpGcsBucketResult.class);
-  }
-
-  private static void assertBucket(
+  private void assertBucket(
       ApiGcpGcsBucketResource actualBucket,
       ApiStewardshipType expectedStewardshipType,
-      String expectedBucketName) {
-    assertEquals(expectedStewardshipType, actualBucket.getMetadata().getStewardshipType());
+      String expectedBucketName,
+      ApiCloningInstructionsEnum expectedCloningInstructions) {
+    assertEquals(workspaceId, actualBucket.getMetadata().getWorkspaceId());
     assertEquals(expectedBucketName, actualBucket.getAttributes().getBucketName());
+    assertEquals(
+        ControlledResourceFixtures.RESOURCE_DESCRIPTION,
+        actualBucket.getMetadata().getDescription());
+    assertEquals(ApiResourceType.GCS_BUCKET, actualBucket.getMetadata().getResourceType());
+    assertEquals(expectedStewardshipType, actualBucket.getMetadata().getStewardshipType());
+    assertEquals(ApiCloudPlatform.GCP, actualBucket.getMetadata().getCloudPlatform());
+    assertEquals(expectedCloningInstructions, actualBucket.getMetadata().getCloningInstructions());
+
+    ApiResourceLineage expectedResourceLineage = new ApiResourceLineage();
```

Comment:
Not for this PR, but my random thought of the day. Right now we generate API objects in flights. This _feels_ wrong. Ideally, we would do all API conversion in the controllers. I think the reason we do this is to allow the JobService to have a generic response return path. I wonder if there is a structure we could implement that would remove the API generation from the flights.